includes/Core/Authentication/Authentication.php

```diff
@@ -432,7 +432,8 @@ private function handle_verification_token() {
 
 		$code = (string) filter_input( INPUT_GET, 'googlesitekit_code' );
 
-		wp_safe_redirect( add_query_arg( 'verify', 'true', $auth_client->get_proxy_setup_url( $code ) ) );
+		// We need to pass the 'missing_verification' error code here so that the URL includes a verification nonce.
+		wp_safe_redirect( add_query_arg( 'verify', 'true', $auth_client->get_proxy_setup_url( $code, 'missing_verification' ) ) );
```

Comment:
This line is getting a bit long. How about this?

```suggestion
		wp_safe_redirect(
			add_query_arg(
				'verify',
				'true',
				$auth_client->get_proxy_setup_url( $code, 'missing_verification' )
			)
		);
```